Added WindowStyle

mrboring opened the pull request:
- #64 

I've attempted to add this myself. I've tested it and it appears to work.

* Added WindowStyle. Defaults to Hidden
* Added to ShellConfig and ExecConfig
* Added unit tests

This is my first attempt at updating a CE! Any feedback is welcome.

CaptnCodr commented:
Thank you for your PR.
I added some comments on the code for consistency.

I would love to see some documentation in the `README.md`. 🙂

mrboring commented:
> I added some comments on the code for consistency. 

Should these be visible on the `Files changed` tab? If yes, I can't see them.

> Also, `Command.fs:35` the line with `WindowStyle = ProcessWindowStyle.Hidden,` can be removed.

Removed.

> I would love to see some documentation in the `README.md`.

I've added an example. I've stated that `WindowStyle` would only work on `Windows`. Is this correct, or would it work on Mac and Linux? If not, is there an `#if`/`#endif` that should be applied?

I'm happy to add more if you think they are needed.

CaptnCodr commented:
> Should these be visible on the Files changed tab? If yes, I can't see them.

Do you see the comments now?

CaptnCodr commented:
In the `README.md` (https://github.com/CaptnCodr/Fli/blob/main/README.md?plain=1#L242) there are these two tables with how to use this CEs.
Below that there are the explainations of the enums and DUs in `Fli` e.g. `Fli.Shells`. You could put `Fli.WindowStyle` cases with the hint that `Hidden` is the default, there.

mrboring commented:
> In the `README.md` (https://github.com/CaptnCodr/Fli/blob/main/README.md?plain=1#L242) there are these two tables with how to use this CEs. Below that there are the explainations of the enums and DUs in `Fli` e.g. `Fli.Shells`. You could put `Fli.WindowStyle` cases with the hint that `Hidden` is the default, there.

Should `Fli.Shells` not be `Fli.Domain.Shells`?

CaptnCodr commented:
> Should Fli.Shells not be Fli.Domain.Shells?

`Domain` is being auto opened, it doesn't matter. It feels like it's directly in `Fli`. 🤫

CaptnCodr commented:
> Is this correct, or would it work on Mac and Linux? If not, is there an #if/#endif that should be applied?

According to microsoft docs, it's supported.

mrboring commented:
I think I've made the requested changes. Please let me know if I've missed anything.

CaptnCodr commented on README.md:
```diff
@@ -252,6 +262,7 @@ cli {
 | `Username`             | `string`                                                 |
 | `Credentials`          | `string * string * string`                               |
 | `WorkingDirectory`     | `string`                                                 |
+| `WindowStyle`          | `Fli.WindowStyle`             |
```
Align this pipe with the other pipes.

mrboring replied:
Fixed.

mrboring commented:
I've made the requested change to README.md.

CaptnCodr commented:
That looks fine. You did good. 🙂 
Thank you for your contribution. ❤️
